File: Тренировка 1.0/01. Сложность, тестирование, особые случаи/G. Детали/Детали.cs
using System;

int n, k, m;
string[] numbers = Console.ReadLine().Split(" ");
n = int.Parse(numbers[0]);
k = int.Parse(numbers[1]);
m = int.Parse(numbers[2]);

int detailsPerBar = k / m;
if (detailsPerBar == 0)
{
    Console.WriteLine(0);
    return;
}
int details = 0;
int detailsWeightPerBar = detailsPerBar * m;
while(n >= k)
{
    int barsCount = n / k;
    n -= barsCount * detailsWeightPerBar;
    details += barsCount * detailsPerBar;
}

Console.WriteLine(details);

File: Тренировка 1.0/01. Сложность, тестирование, особые случаи/G. Детали/Детали.cpp
#include <iostream>

using namespace std;

int main()
{
    int n, k, m;
    cin >> n >> k >> m;
    if (k / m == 0)
    {
        cout << 0;
        return 0;
    }
    int details = 0;
    while (n >= k)
    {
        details += k / m;
        n += k % m - k;
    }

    cout << details;
}